src/Components/Forms/src/EditContextDataAnnotationsExtensions.cs

```diff
@@ -35,8 +35,19 @@ public static EditContext AddDataAnnotationsValidation(this EditContext editCont
     /// <returns>A disposable object whose disposal will remove DataAnnotations validation support from the <see cref="EditContext"/>.</returns>
     public static IDisposable EnableDataAnnotationsValidation(this EditContext editContext)
     {
-        return new DataAnnotationsEventSubscriptions(editContext);
+        return new DataAnnotationsEventSubscriptions(editContext, null);
     }
+    /// <summary>
+    /// Enables DataAnnotations validation support for the <see cref="EditContext"/>.
+    /// </summary>
+    /// <param name="editContext">The <see cref="EditContext"/>.</param>
+    /// <param name="serviceProvider">The <see cref="IServiceProvider"/> to be used in the <see cref="ValidationContext"/>.</param>
+    /// <returns>A disposable object whose disposal will remove DataAnnotations validation support from the <see cref="EditContext"/>.</returns>
+    public static IDisposable EnableDataAnnotationsValidation(this EditContext editContext, IServiceProvider? serviceProvider)
```

Comment:
@SteveSandersonMS how do you feel about using `EditContext.Properties` to pass this along? It does make it harder to discover, but it avoids a new API. I don't think there is a scenario where `EditContext` and `IServiceProvider` have different lifetimes.

Reply:
@pranavkm The user is allowed to create their own ``EditContext`` to supply it to an ``<EditForm>``.  In those cases, you don't have control over its lifetime.

Reply:
I guess, but presumably they aren't using a static instance of `EditContext` that's shared across multiple Blazor circuits which would cause it to have a longer lifetime than the scoped IServiceProvider. But perhaps you're right, not collating the two is the right way to approach it.

Reply:
I'm also concerned about having these two separate overloads. I wonder if it might be better overall to consider a breaking change here. At the very least I think we should mark as obsolete the older overload, and have the new overload explicitly throw if you try to pass a null `serviceProvider`.

If existing code continues not to supply a service provider, then it's going to be super hard for people to reason about why in certain cases their validators can't resolve a DI service, and in other cases they can.

These are relatively uncommon APIs so it's not many people who will be inconvenienced, but I'd rather them be inconvenienced by an "obselete API" warning than by a nullref from their validator logic.

Reply:
```suggestion
    public static IDisposable EnableDataAnnotationsValidation(this EditContext editContext, IServiceProvider serviceProvider)
```